__tests__/implementation.test.js

```diff
@@ -1217,3 +1217,235 @@ test('bail if errorHandler does not return an error', async () => {
+        ]);
+    });
+});
+
+test('batch endpoint (multiple requests) with propertyBatchKey error handling', async () => {
+    const config = {
+        resources: {
+            foo: {
+                isBatchResource: true,
+                docsLink: 'example.com/docs/bar',
+                batchKey: 'foo_ids',
+                newKey: 'foo_id',
+                propertyBatchKey: 'properties',
+                responseKey: 'id',
+            },
+        },
+    };
+
+    const resources = {
+        foo: ({ foo_ids, properties, include_extra_info }) => {
+            if (_.isEqual(foo_ids, [2, 4, 5])) {
+                expect(include_extra_info).toBe(true);
+                return Promise.resolve([
+                    {
+                        id: 2,
+                        name: 'Burger King',
+                        rating: 3,
+                    },
+                    {
+                        id: 4,
+                        name: 'In N Out',
+                        rating: 3.5,
+                    },
+                    {
+                        id: 5,
+                        name: 'Shake Shack',
+                        rating: 4,
+                    },
+                ]);
+            }
+            if (_.isEqual(foo_ids, [1, 3])) {
+                expect(include_extra_info).toBe(false);
+                throw new Error('yikes');
+            }
+        },
+    };
+
+    await createDataLoaders(config, async (getLoaders) => {
+        const loaders = getLoaders(resources);
+
+        const results = await loaders.foo.loadMany([
+            { foo_id: 1, properties: ['name', 'rating'], include_extra_info: false },
+            { foo_id: 2, properties: ['name', 'rating'], include_extra_info: true },
+            { foo_id: 3, properties: ['name'], include_extra_info: false },
+            { foo_id: 4, properties: ['rating'], include_extra_info: true },
+            { foo_id: 5, properties: ['name'], include_extra_info: true },
+        ]);
+
+        expect(results).toEqual([
+            expect.toBeError(/yikes/),
+            { id: 2, name: 'Burger King', rating: 3 },
+            expect.toBeError(/yikes/),
+            { id: 4, rating: 3.5 },
+            { id: 5, name: 'Shake Shack' },
+        ]);
+    });
+});
+
+test('batch endpoint with propertyBatchKey with reorderResultsByKey handles response with non existant items', async () => {
+    const config = {
+        resources: {
+            foo: {
+                isBatchResource: true,
+                docsLink: 'example.com/docs/bar',
+                batchKey: 'foo_ids',
+                newKey: 'foo_id',
+                reorderResultsByKey: 'foo_id',
+                propertyBatchKey: 'properties',
+                responseKey: 'foo_id',
+            },
+        },
+    };
+
+    const resources = {
+        foo: ({ foo_ids, bar }) => {
+            if (_.isEqual(foo_ids, [1, 2, 3])) {
+                return Promise.resolve([
```

Comment:
```suggestion
            if (_.isEqual(foo_ids, [1, 2, 3])) {
                expect(bar).toBe(true);
                return Promise.resolve([
```

i assume?

Reply:
I'm assuming "bar" is supposed to serve a similar purpose to include_extra_info - i.e. to make a different chunk and test multiple request groups?